public/app/features/dashboard/components/SaveDashboard/SaveDashboardErrorProxy.tsx

```diff
@@ -50,7 +51,7 @@ export const SaveDashboardErrorProxy = ({
           onDismiss={onDismiss}
         />
       )}
-      {error.data && error.data.status === 'name-exists' && (
+      {error.data && error.data.status === 'name-exists' && !isRestoreDashboardsEnabled && (
```

Comment:
can we make this so there's no possibility of accidentally rendering both?
couple of options:
- wrap in a jsx fragment and use a ternary
```tsx
{error.data && error.data.status === 'name-exists' && (
  <>
    {isRestoreDashboardsEnabled ? (
      <Modal ... />
    ) : (
      <ConfirmModal ... />
    )}
  </>
)}
```
- create a variable further up to hold the `nameExistsErrorComponent`
```tsx
const nameExistsErrorComponent = isRestoreDashboardsEnabled ? <Modal ... /> : <ConfirmModal ... />

...

{error.data && error.data.status === 'name-exists' && nameExistsErrorComponent}
```